src/session_impl.cpp

```diff
@@ -1685,8 +1691,13 @@ namespace aux {
 					!= m_settings.get_str(settings_pack::listen_interfaces));
 
 		apply_pack(&pack, m_settings, this);
+
 		m_disk_thread.set_settings(&pack, m_alerts);
 
+#if TORRENT_USE_INVARIANT_CHECKS
+		check_invariant();
```

Comment:
it's not obvious to me why this should be called here. this function is an internal function (it ends with ``_impl``), so I would expect whatever calls this to enforce the invariant.